src/main/scala/sangria/execution/deferred/FetcherBasedDeferredResolver.scala

```diff
@@ -78,15 +78,36 @@ class FetcherBasedDeferredResolver[-Ctx](fetchers: Vector[Fetcher[Ctx, _, _, _]]
 
     val (nonCachedIds, cachedResults) = partitionCachedRel(ctx.cache, relIds)
 
-    val newResults =
-      if (nonCachedIds.nonEmpty)
-        f.fetchRel(ctx, RelationIds(nonCachedIds.asInstanceOf[Map[Relation[Any, _, _], Seq[Any]]]))
-          .map(groupAndCacheRelations(ctx, nonCachedIds, _))
-      else
-        Future.successful(MutableMap.empty[Relation[Any, Any, Any], MutableMap[Any, Seq[Any]]])
+    val groupedRelIds = ctx.fetcher.config.maxBatchSizeConfig match {
+      case Some(size) => nonCachedIds.map { case (rel, ids) => (rel, ids.grouped(size))}
+      case None => nonCachedIds.map { case (rel, ids) => (rel, Iterator(ids)) }
```

Comment:
To avoid the cost of calling a method with var-args:
```suggestion
      case None => nonCachedIds.map { case (rel, ids) => (rel, Iterator.single(ids)) }
```